packages/process/src/process/env/node-env.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@apoyo/scopes'
+import { Enum, Err, Str } from '@apoyo/std'
+
+export enum NodeEnvironment {
+  DEV = 'development',
+  STAGING = 'staging',
+  PROD = 'production',
+  TEST = 'test'
+}
+
+const isValidEnv = Str.oneOf(Enum.values(NodeEnvironment))
+
+export const $nodeEnv = Injectable.define<NodeEnvironment>(() => {
+  const nodeEnv = process.env.NODE_ENV
+  if (!nodeEnv) {
+    return NodeEnvironment.PROD
+  }
+  if (isValidEnv(nodeEnv)) {
```

Comment:
should the library restrict the possible nodeEnv values?

Reply:
no => add predefined values and environments, but the library should not restrict the possibles values.